docs/model-dev-guide/batch-processing/batch-process-api-ug.rst

```diff
@@ -0,0 +1,173 @@
+.. _torch-batch-processing-ug:
+
+############################
+ Torch Batch Processing API
+############################
+
+.. meta::
+   :description: Learn how to use the Torch Batch Processing API.
+
+In this guide, you'll learn about the :ref:`Torch Batch Process API <torch-batch-process-api-ref>`
+and how to perform batch inference (also known as offline inference).
+
++---------------------------------------------------------------------+
+| Visit the API reference                                             |
++=====================================================================+
+| :ref:`torch-batch-process-api-ref`                                  |
++---------------------------------------------------------------------+
+
+.. caution::
+
+   This is an experimental API and may change at any time.
+
+**********
+ Overview
+**********
+
+The Torch Batch Processing API takes in (1) a dataset and (2) a user-defined processor class and
+runs distributed data processing.
+
+This API automatically handles the following for you:
+
+-  shard a dataset by number of workers available
+-  apply user-defined logic to each batch of data
+-  handle synchronization between workers
+-  track job progress to enable preemption and resumption of trial
+
```

Comment:
This API allows you to accomplish the following tasks:

or

This API automatically handles the following for you:

-  shards a dataset by number of workers available
-  applies user-defined logic to each batch of data
-  handles synchronization between workers
-  tracks job progress to enable preemption and resumption of trial